torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -624,9 +624,15 @@ def _sharded_pre_load_state_dict_hook(
         tensor = torch.empty(
             chunk_size * fsdp_state.world_size, dtype=local_tensor.dtype, device=device
         )
-        dist.all_gather_into_tensor(
-            tensor, local_tensor, group=fsdp_state.process_group
-        )
+        if local_tensor.is_cpu:
+            tensor_list = list(
```

Comment:
I think there might be a corner case that needs to be taken care. 
Say we are chunking tensor of size 3 in world size 4, the tensor_list would be length of 3, dist.all_gather() would fail since len(tensor_list != dist.get_world_size()), right? Correct me if I am wrong.

Reply:
yes, it will probably blow up, we should add this hack to c10d cuz I copy-pasted this snipped a few times already